### ISSUE TITLE
Blazor wasm base path should be configurable post publish

### ISSUE THREAD

**Issue by ndc** [dotnet/aspnetcore#22148]
Currently the only way I can find to set Blazor wasm base path is by editing index.html `<base href="/thepath/" />` and then publish it.

If I change the base path in the generated index.html after publish, when running the app I will get an error:

> Failed to find a valid digest in the ‘integrity’ attribute for resource ‘http://mydomain/app/mybwa/_framework/wasm/dotnet.3.2.0.js’ with computed SHA-256 integrity ‘blabla…’. The resource has been blocked.

Suppose I want to run 5 different instances of the same app in different folders (for example shared hosting), I need to edit index.html and publish five times.

Alternatively I can [disable integrity check](https://github.com/dotnet/aspnetcore/issues/19828#issuecomment-601823319) so I can freely edit index.html post publish, but I'm not sure about the side effects.

In ASP.Net Core we can set `app.UsePathBase` with a value from IConfiguration.

It would be good if blazor wasm can read base path from configuration too.

Or is there a way to recalculate the signature for integrity check? Or to exclude index.html from integrity check?

**Comment by nvmkpk** [comment]
You can easily achieve this if you are hosting the blazor application in asp.net core application. With that, you can bypass index.html and configure a cshtml file where you can set base path in the you would like to.

**Comment by ndc** [comment]
@nvmkpk point taken, but then you need a server that can run executables. Would be nice if base path is configurable and the app runs on static file server.

**Comment by nvmkpk** [comment]
True. But then how do you expect to configure the base path in a static file if you cannot run any code in server?

**Comment by ndc** [comment]
As I said in the post, I can already do this by editing index.html with a text editor, changing the base href, and then deploy (copy to server), provided the BlazorCacheBootResources in csproj is set to false.

Another thing that I have tried: having `appsettings.json` containing base path as one of the config. But I couldn't find a way to update `builder.HostEnvironment.BaseAddress` in Program.Main because it is read only. Even if this is possible I'm not sure that I can edit `appsettings.json` post deploy without turning off integrity check.

**Comment by Swimburger** [comment]
I 100% agree there should be an easier way to change the base path without causing integrity issues.
I'm changing the base-tag after the publish, but I don't run into the same integrity issue. The change in index.html should not affect the integrity of _dotnet.3.2.0.js_. AFAIK, changing index.html would only cause integrity issues when using the PWA feature ([blog post](https://swimburger.net/blog/dotnet/fix-blazor-webassembly-pwa-integrity-checks)).

My guess is that something else is causing the integrity issue. I have a [working Blazor WASM sample](https://swimburger.github.io/BlazorGitHubPagesDemo/) hosted in a subfolder on GitHub Pages.

**Comment by ndc** [comment]
Hi @Swimburger thanks for the info.
I have recently upgraded my blazor app to 5.0, but I hadn't tried removing the BlazorCacheBootResources tag. After reading your comment I tried to remove the tag, and my app works even with modified index.html. So now signature checking is active and I can still update the base path. Jubilations!

Between 3.2.0 and 5.0.1 someone decided to not include the `index.html` in `blazor.boot.json` and that someone has my thanks.

This ticket can be closed now.
